content/docs/engine-dev/tools/tiaf/maintaining-tiaf/debugging-tiaf.md

```diff
@@ -77,47 +87,182 @@ In the log above, you can extract the following information about the AR run and
+                "--dst-branch", "development",
+                "--commit", "85a0bd96daa4bd330d56bd817e8cdaf29d172b63",
+                "--build-number", "001", // The build number can be anything
+                "--suites", "smoke", "main", // The suites must match those used in AR
+                "--label-excludes", "REQUIRES_gpu", // The label exclude filters must match those used in AR
+                "--test-failure-policy", "continue",
+                "--target-output", "stdout",
+                "--runtime-type", "native" // change to "python" for Python TIAF
+               ]
+        }
+    ]
+}
+```
+
+When the script has completed execution you will see the same result as the failing AR run (in this instance, a crash) as well as the change list and arguments for the runtime.
+
+{{< note >}}
+This step need only be performed once and the change list and arguments reused as needed.
+{{< /note >}}
+
+### Step 3: Run the TIAF runtime with a debugger attached
+
+In the previous step, the TIAF log output states the command line options used by the runtime:
+
+```
+[2023-04-30 00:34:45,577][TIAF][INFO] Args: --sequence=tianowrite --fpolicy=continue --suites=main,smoke --labelexcludes=REQUIRES_gpu --ipolicy=continue --changelist=C:\dev\o3de\build\windows_vs2019\bin\TestImpactFramework\debug\Temp\Native\changelist.b9ad24eef8544d43a9d3fa75912532d4.json --report=C:\dev\o3de\build\windows_vs2019\bin\TestImpactFramework\debug\Temp\Native\Reports\report.b9ad24eef8544d43a9d3fa75912532d4.json --targetout=stdout
+```
+
+Thus, the arguments we need to supply to the runtime in your IDE are as follows:
+
+```
+--sequence=tianowrite --fpolicy=continue --suites=main,smoke --labelexcludes=REQUIRES_gpu --ipolicy=continue --changelist=C:\dev\o3de\build\windows_vs2019\bin\TestImpactFramework\debug\Temp\Native\changelist.b9ad24eef8544d43a9d3fa75912532d4.json --report=C:\dev\o3de\build\windows_vs2019\bin\TestImpactFramework\debug\Temp\Native\Reports\report.b9ad24eef8544d43a9d3fa75912532d4.json --targetout=stdout
+```
+
+In your IDE, find the appropriate console front end project for the AR run (either `TestImpact.Frontend.Console.Native` or `TestImpact.Frontend.Console.Python`) and provide the arguments above. Now run the project with your debugger attached to find and fix the issue causing the AR failure.
+
+For this AR run, the cause of the crash was determined to be a bug due to an edge case where a test target had previously been opted-in to the TIAF (whereupon its coverage data was stored in the `historic_data.json` file used by the run), only for that test target to have been subsequently opted-out at a later date, causing a `nullptr` dereference that was fixed in commit `98dbed226cdae5afa5da25fe4bf84ae6760e4ac4`.
+
+{{< tip >}}
+You will save a lot of time by reproducing issues with the TIAF itself locally rather than attempting to debug them in AR with the limited tooling available.
+{{< /tip >}}
+
+### Running failing tests
```

Comment:
@jonawals Can you verify if this header more accurately describes the purpose of this section?
"Running failing tests", I feel like this is what we've been doing in the prior steps. I understand this section to be for the case where we want to run a specific test target that's failed.
```suggestion
### Running specific test targets
```